### DISCUSSION TITLE
Q: "Best Practices"

### DISCUSSION THREAD

**Discussion by zz0rk** [garethgeorge/backrest#203] [General]
https://github.com/garethgeorge/backrest/wiki#best-practices

> Best Practices
> Configure a reasonable retention policy for each plan. Backrest performs well up to a history of ~1000s of snapshots but too many may eventually slow performance.
> Backup your configuration (e.g. $BACKREST_CONFIG or $HOME/.config/backrest/config.json by default on Linux/MacOS)
> Your configuration contains the encryption keys for your repositories. If you loose this file you will not be able to restore your backups.
> You may alternatively backup your encryption keys individually in which case you will be able to use restic directly to restore your backups.

1. I can't find the backrest configuration file at ~/.config/backrest/config.json. I run Debian 12.
Also: $ whereis backrest
backrest: /usr/local/bin/backrest
 
2. You mention the encryption key/s. Do you mean what the restic documentation refers to as the "repo password"? I just want to make sure I am not missing something.

BTW, fantastic tool and great name!

Thank you

**Comment by garethgeorge** [comment]
Hey, glad Backrest is working well for you!

Re: the config.json, make sure you're checking the home directory of the user running backrest (is it possible you installed with sudo? Backrest may be running as root e.g. `/root/.config/backrest/config.json`

Find the systemd config that you added to be sure.

Re: encryption keys, yep this is the repo password! Make sure you have that saved somewhere at a minimum as with the repo password and the URI you can always add the repo to another backrest installation and restore from there. I still recommend backing up the whole config as it's convenient and will retain your configured backup plans, etc.

**Reply by zz0rk** [reply]
@garethgeorge Yeah, I installed it as root, :facepalm: 
Thanks for clarifying my encryption keys question.
Thanks again for this great tool.
